rust-arroyo/src/backends/kafka/producer.rs

```diff
@@ -243,4 +243,20 @@ mod tests {
             .produce(&destination, payload)
             .expect("Message produced")
     }
+
+    #[test]
+    #[ignore = "This should fail with a KafkaError but doesn't for some reason"]
```

Comment:
I could not figure out how to get the produce to fail. Even with a non-existent topic, referencing a partition that doesn't exist, on a broker that doesn't exist, this test still passes.

Reply:
Ah yes, @fpacifici @bmckerry found an issue with these producers: they do not await until the produce is actually done. see how the arroyo produce waits with callback+future, that's missing. maybe you can fix it in the same PR? otherwise it goes to the backlog

Reply:
Ah OK I'll see if I can fix that.